src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -109,7 +109,7 @@ public static void ApplyDefaultsTo(Config config)
                 new List<string>(),
                 defaultTag: "alpha",
                 defaultIncrementStrategy: IncrementStrategy.Minor,
-                defaultVersioningMode: VersioningMode.ContinuousDeployment,
+                defaultVersioningMode: config.VersioningMode == VersioningMode.Mainline? VersioningMode.Mainline : VersioningMode.ContinuousDeployment,
```

Comment:
Why can't `defaultVersioningMode` always be set to `config.VersioningMode`?

Reply:
Because documentation says that the develop branch is set to ContinuousDeployment mode by default. And all tests require develop branch in ContiniousDeployment mode. But in mainline mode we get wrong behaviour.

Reply:
Aha, that makes sense.